packages/@aws-cdk/cdk/lib/cloudformation/resource.ts

```diff
@@ -93,23 +112,83 @@ export class Resource extends Referenceable {
         this.dependsOn.push(...other);
     }
 
+    /**
+     * Adds an override to the synthesized CloudFormation resource. To add a
+     * property override, either use `addPropertyOverride` or prefix `path` with
+     * "Properties." (i.e. `Properties.TopicName`).
+     *
+     * @param path  The path of the property, you can use dot notation to
+     *              override values in complex types. Any intermdediate keys
+     *              will be created as needed.
+     * @param value The value. Could be primitive or complex.
+     */
+    public addOverride(path: string, value: any) {
+        const parts = path.split('.');
+        let curr: any = this.rawOverrides;
+
+        while (parts.length > 1) {
+            const key = parts.shift()!;
+
+            // if we can't recurse further or the previous value is not an
+            // object overwrite it with an object.
+            const isObject = curr[key] != null && typeof(curr[key]) === 'object' && !Array.isArray(curr[key]);
+            if (!isObject) {
+                curr[key] = { };
+            }
+
+            curr = curr[key];
+        }
+
+        const lastKey = parts.shift()!;
+        curr[lastKey] = value;
+    }
+
+    /**
+     * Syntactic sugar for `addOverride(path, undefined)`.
+     * @param path The path of the value to delete
+     */
+    public addDeletionOverride(path: string) {
+        this.addOverride(path, undefined);
+    }
+
+    /**
+     * Adds an override to a resource property.
+     *
+     * Syntactic sugar for `addOverride("Properties.<...>", value)`.
+     *
+     * @param propertyPath The path of the property
+     * @param value The value
+     */
+    public addPropertyOverride(propertyPath: string, value: any) {
```

Comment:
Is it honestly even worth having these? The syntactic sugar they offer is trivial.